### ISSUE TITLE
PowerShell installation fails on Windows 7 (PowerShell 2.0) with "Invoke-WebRequest is not recognized"

### ISSUE THREAD

**Issue by realityexists** [filipw/dotnet-script#175]
Run

> (new-object Net.WebClient).DownloadString("https://raw.githubusercontent.com/filipw/dotnet-script/master/install/install.ps1") | iex

on Windows 7 Pro SP1. An error occurs:

> The term 'Invoke-WebRequest' is not recognized as the name of a cmdlet, function, script file, or operable program. Check the spelling of the name, or if a path was included, verify that the path is correct and try again.

**Comment by seesharper** [comment]
Could you try to bump your PowerShell version?
https://stackoverflow.com/questions/32315616/the-term-invoke-webrequest-is-not-recoginzed-as-the-name-of-a-cmdlet

**Comment by realityexists** [comment]
Upgraded to PowerShell 4.0 and now I get a different error:

> Microsoft.PowerShell.Archive\Expand-Archive : The module 'Microsoft.PowerShell.Archive' could not be loaded. For more information, run 'Import-Module Microsoft.PowerShell.Archive'.
> At line:19 char:1
> + Microsoft.PowerShell.Archive\Expand-Archive $zipFile -DestinationPath $installat ...
> + ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
>     + CategoryInfo          : ObjectNotFound: (Microsoft.Power...\Expand-Archive:String) [], CommandNotFoundException
>     + FullyQualifiedErrorId : CouldNotAutoLoadModule
>

**Comment by seesharper** [comment]
Could you try installing the Microsoft.PowerShell.Archive module?
https://www.powershellgallery.com/packages/Microsoft.PowerShell.Archive/1.1.0.0

**Comment by realityexists** [comment]
> Install-Module -Name Microsoft.PowerShell.Archive
> The term 'Install-Module' is not recognized as the name of a cmdlet

**Comment by realityexists** [comment]
Tried again on another Windows 7 (x64) machine, this time installing PowerShell 5.1. Now I get:

```
Downloading latest verson (0.15.0)
You cannot call a method on a null-valued expression.
At line:24 char:1
+ $paths = $path.Split(";") -inotlike "*dotnet.script*"
+ ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : InvalidOperation: (:) [], RuntimeException
    + FullyQualifiedErrorId : InvokeMethodOnNull

```

**Comment by seesharper** [comment]
We would very much appreciate if you could take the time to debug this. Currently without access to a Windows 7 machine. If you install the PowerShell extension in VS Code it should be fairly easy so debug. Otherwise we provide a Chocolatey install as well. https://chocolatey.org/packages/dotnet.script

**Comment by realityexists** [comment]
OK, debugged and it actually worked in the debugger. The line it was failing on, though, was 

     $paths = $path.Split(";") -inotlike "*dotnet.script*" 

I think it failed, because I had the no PATH set as a user environment variable, so the previous line:

     $path = [System.Environment]::GetEnvironmentVariable("path", [System.EnvironmentVariableTarget]::User);

set $path to null.

Anyway, it now seems to have added itself several times. My user PATH is now

    C:\ProgramData\dotnet-script\dotnet-script;C:\ProgramData\dotnet-script\dotnet-script;C:\ProgramData\dotnet-script\dotnet-script

**Comment by seesharper** [comment]
What if you change this line 
https://github.com/filipw/dotnet-script/blob/60de06f07cfc4e4be34a5cef7168741126d00acc/install/install.ps1#L24

into 
```
$paths = $path.Split(";") -inotlike "*dotnet-script*"  
```

Does it work as expected then?

**Comment by realityexists** [comment]
Yes, that seems to prevent it from adding a duplicate entry to the path.
